src/Tribe/Tickets.php

```diff
@@ -4639,6 +4640,110 @@ public function maybe_setup_attendee_user_from_email( $email, $args = [] ) {
 			return $created;
 		}
 
+		/**
+		 * Returns either the post IDs or the number of orphaned posts based on the provider.
+		 *
+		 * @since TBD
+		 *
+		 * @param bool $count Whether we should return the post IDs (default, false) or the number of posts found (true).
+		 *
+		 * @return array|int Either the post IDs of the orphaned posts (default, $count = false) or the number of orphaned posts ($count = true).
+		 */
+		public function get_orphaned_posts( bool $count = false ) {
+			$provider          = static::class;
+			$orphaned_post_ids = $this->get_orphaned_post_ids( $provider );
+
+			// If counting, return the numbers.
+			if ( $count ) {
+				return count( $orphaned_post_ids );
+			}
+
+			// Return the post IDs to be deleted.
+			return $orphaned_post_ids;
+		}
+
+		/**
+		 * Query to get the orphaned entries of the given post type.
+		 *
+		 * @since TBD
+		 *
+		 * @param string $provider The provider to check for orphaned posts ('rsvp' or 'tc_ticket').
+		 *
+		 * @return array Array of orphaned post IDs.
+		 */
+		public function get_orphaned_post_ids( $provider ): array {
+			global $wpdb;
+
+			// Check for cached results first.
+			$cache_key       = 'tec_tickets_orphaned_posts_' . sanitize_key( $provider );
+			$cached_post_ids = get_transient( $cache_key );
+
+			if ( is_array( $cached_post_ids ) ) {
+				/**
+				 * Filter the list of orphaned post IDs for a specific provider.
+				 *
+				 * @since TBD
+				 *
+				 * @param array      $cached_post_ids Array of orphaned post IDs from cache.
+				 * @param string     $provider        The provider being checked.
+				 * @param bool       $cached          Whether the results are from cache (true) or fresh query (false).
+				 * @param array|null $meta_keys       Meta keys used in the query (null for cached results).
+				 */
+				return (array) apply_filters( 'tec_tickets_orphaned_post_ids', $cached_post_ids, $provider, true, null );
```

Comment:
You have cached them while already filtered.

So either cache them un-filtered and filter them in both places or don't filter them here

Reply:
Done.